sdk-docs/project-structure.md

```diff
@@ -0,0 +1,93 @@
+- `validDomains`: A list of any external domains that the app needs to communicate with. The list is audited in the approval step of the app and every external API call must be listed here to get through the verification process.
+- `validForAccountIds`: Put `ALL_ACCOUNTS` in if your app is available to all Trackunit users. If you want to only make it available to certain users, provide a list of account ID's as strings.
+- `activationMode`: Controls how the app can/will be installed.
+    - Enabled means it's enabled on the subscription when this Iris app is approved.
+    - Visible means it will be visible to the customer to choose for them selves to enable/disable it.
+    - FORCE_ENABLE means it cannot be removed by the endcustomer, and will be installed based on the selected subscription package.
+    - DEFAULT_ENABLE means it will be enabled when a customer activates a new subscription package.
+    - ALWAYS_VISIBLE means it will always be visible for all customers.
+    - CONTROLLED_MANUAL_INSTALLATION means it will be controlled by the team that activates the subscription package for the customer, not activating the app.
+    - CONTROLLED_AUTOMATIC_INSTALLATION means it will be controlled by the team that activates the subscription package for the customer, also adding.
+    - All modes will take into consideration both **validForAccountIds** and **marketplace.allowForPackage**
+- `marketplace`: Holds all the settings associated with your apps listing on the marketplace.
+    - `name`: Well..the name of your app.
+    - `description`: Write a small summary of your app here that describes what it does.
+    - `fullDescriptionPath`: The path to a markdown file that explains your app in more detail. Store the markdown file somewhere in your project structure as you see fit.
+    - `allowForPackage`: Defines with which subscription packages your app is compatible. Take great care when entering this, as your app might bug out (if it went through the approval phase in the first place) if what is stated here does not match reality.
+    - `tags`: A list of relevant tags to make your app easier to find.
+    - `categories`: A list of categories that your app falls under.
+- `extensions`: What extensions does this app install?
+
+It is a good idea to inspect the Typescript interface called **IrisAppManifest** in the code to get a good idea on all the configurations you can set and what it means.
+### nx.json
+nx.json is the main configuration file for the Nx workspace, containing project-specific settings, cache settings, and task runner configurations.
+
+### tsconfig.json
+There's rarely any reason to change these. It is a configuration file used by the TypeScript compiler. It contains various settings and options
+that determine how the TypeScript compiler should transpile your TypeScript code into JavaScript, 
+as well as how it should handle type checking and other features.
```

Comment:
Consider making this very brief and linking to the TypeScript docs or NX config (to explain the .lib version). Developers should be familiar with this file if they have ever worked with TypeScript. 😊

Reply:
Note taken! Will try to improve it a bit.
Generally I would like for our docs to be as self-contained as possible, so the dev can have a streamlined and smooth onboarding of our tech. If I reference external sources too much, they'll have to context switch a lot going back and forth and we might lose their attention.

This doesn't mean that I'll redocument all of typescript and all of nx into our docs, but if I can say a sentence of two and tell the developer (even those who never worked with TS or NX) **very** briefly what things mean and do, then I try to do that :)